src/GitVersion.Core/VersionCalculation/NextVersionCalculator.cs

```diff
@@ -11,21 +12,23 @@ public class NextVersionCalculator : INextVersionCalculator
     private readonly IBaseVersionCalculator baseVersionCalculator;
     private readonly IMainlineVersionCalculator mainlineVersionCalculator;
     private readonly IRepositoryStore repositoryStore;
+    private readonly IIncrementStrategyFinder incrementStrategyFinder;
     private readonly Lazy<GitVersionContext> versionContext;
     private GitVersionContext context => this.versionContext.Value;
 
     public NextVersionCalculator(ILog log, IBaseVersionCalculator baseVersionCalculator,
         IMainlineVersionCalculator mainlineVersionCalculator, IRepositoryStore repositoryStore,
-        Lazy<GitVersionContext> versionContext)
+        IIncrementStrategyFinder incrementStrategyFinder, Lazy<GitVersionContext> versionContext)
```

Comment:
Can you please just format this constructor so every argument sits on its own line? 

We should also do something about the number of arguments here, as everything more than 3 is too much, imho. Not in this PR, though. :)

Reply:
Totally agree with you. In my refactoring of v6 I already removed some calls to IRepositoryStore. In my opinion this dependency needs to be removed:

1. IRepositoryStore need to be removed because it is to low level and not necessary
2. The logic of IBaseVersionCalculator will be merged to NextVersionCalculator.
3. GitVersionContext should be not created in the DI factory. This class holds context related properties with a dedicated scope in the NextVersionCalculator class.
4.  IIncrementStrategyFinder is also obsolete in long-term in my opinion. It makes no sense because we have more than one increment strategy in v6.
5. IMainlineVersionCalculator is also obsolete in long-term in my opinion. It makes no sense to have a business logic for a dedicated mode. This gives to much if then else. Better would be to write dedicated IVersionStrategy implementation and make it configurable via the configuration file.

Reply:
> 5\. IMainlineVersionCalculator is also obsolete in long-term in my opinion. It makes no sense to have a business logic for a dedicated mode. This gives to much if then else. Better would be to write dedicated IVersionStrategy implementation and make it configurable via the configuration file.

I agree with this

Reply:
I think I agree with everything. Especially point 5 where I think each `mode` should give a different `Strategy` that all use the same `Calculator` implementations in different ways depending on how stuff is configured. So a `Strategy` basically just serves as a default template of configuration and then every key can be overridden in `GitVersion.yml`, resulting in a custom strategy based on the selected strategy as defined with `mode`. I hope that makes sense. :)

Reply:
Done